components/client/src/test/unit/java/com/hotels/styx/client/netty/NettyToStyxResponsePropagatorTest.java

```diff
@@ -268,6 +272,27 @@ public void shouldReleaseAlreadyReadBufferInCaseOfError() throws Exception {
     }
 
     @Test
+    public void mapsOutOfDirectMemoryExceptionsToResourceExhaustedException() throws Exception {
+        EmbeddedChannel channel = new EmbeddedChannel(new NettyToStyxResponsePropagator(responseSubscriber, SOME_ORIGIN));
+        channel.writeInbound(new DefaultHttpResponse(HTTP_1_1, OK));
+
+        channel.pipeline().fireExceptionCaught(newOutOfDirectMemoryError("Simulated out of direct memory error in a test."));
+
+        assertThat(responseSubscriber.getOnErrorEvents().get(0), is(instanceOf(StyxClientError.class)));
+    }
+
+    private OutOfDirectMemoryError newOutOfDirectMemoryError(String message) throws InstantiationException, IllegalAccessException, java.lang.reflect.InvocationTargetException {
```

Comment:
Perhaps a brief comment to explain that this is because OutOfDirectMemoryError has package-private visibility.
Also, `InvocationTargetException` does not need to be fully-qualified.